src/kbc_modules/cc_kbc/attester/az_snp_vtpm/mod.rs

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) 2023 Microsoft Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use super::Attester;
+use anyhow::*;
+use az_snp_vtpm::vtpm;
+use serde::{Deserialize, Serialize};
+
+pub fn detect_platform() -> bool {
+    vtpm::has_tpm_device()
+}
+
+#[derive(Debug, Default)]
+pub struct AzSnpVtpmAttester;
+
+#[derive(Serialize, Deserialize)]
+struct Evidence {
+    quote: vtpm::Quote,
+    report: Vec<u8>,
+}
+
+impl Attester for AzSnpVtpmAttester {
+    fn get_evidence(&self, report_data: String) -> Result<String> {
+        let report = vtpm::get_report()?;
+        let report_data_bin = base64::decode(&report_data)?;
+        let quote = vtpm::get_quote(&report_data_bin)?;
+
+        let evidence = Evidence { quote, report };
```

Comment:
Are the VCEK, ASK, ARK sent here? 
Or do we expect the verifier to extract the chip ID and fetch the VCEK and chain from AMD's keys service?

Reply:
We can fetch VCEK/ASK/ARK from the IMDS server on the underlying node and i think that would work better than the KBS having to reach out to AMD's servers.
See also: https://github.com/mkulke/attestation-service/pull/1#discussion_r1139900623.

Reply:
At the moment the verifier does fetch the VCEK + cert chain from AMD, which is not practical (there are rather rigid rate limits on that server).

There are multiple options:
- A metadata endpoints returns VCEK, ASK, ARK in Azure, the attester could include them in the report (extended snp report)
- The ARK + ASK could be part of the compiled binary (I think the nascent https://github.com/virtee/snp crate bundles the ARKs and ASK for genoa and milan)
- There is a caching facility in the sev crate (`sev::cached_chain`)

I'm leaning towards bundling VCEK, ASK in the report from the attester and caching the ARK on the verifier certs.

Reply:
For reference the draft of the SNP attester/verifier uses the extended report to hold the VCEK (cached on the host, basically) and the ARK and ASK are in the verifier source.

Reply:
I think we'll do something similar. We'll fetch the VCEK from the metadata service (and cache it) and use the key chain bundled in the sev crate.